implementations/rust/ockam/ockam_node/src/context/context_lifecycle.rs

```diff
@@ -115,8 +116,38 @@ impl Context {
 
     /// Utility function to sleep tasks from other crates
     #[doc(hidden)]
-    pub async fn sleep(&self, dur: Duration) {
-        tokio::time::sleep(dur).await;
+    pub async fn sleep(&self, duration: Duration) {
+        tokio::time::sleep(duration).await;
+    }
+
+    /// Utility function to sleep tasks for long periods of time (seconds precision)
+    /// Difference between this and `sleep` is that this sleeps in 1 second intervals and recalculates time left,
+    /// which account for the time the device was in sleep state
+    #[doc(hidden)]
+    pub async fn sleep_long_until(&self, deadline_timestamp_seconds: u64) {
```

Comment:
While we're here I would move the `TimestampInSeconds` data type from `ockam_identity` to `ockam_node`:

-  It's not a concern specifically tied to Identity
- This way we can give a clearer type to the `deadline_timestamp` parameter